server/build/webpack/base.config.js

```diff
@@ -0,0 +1,318 @@
+          ]
+        },
+        {
+          test: /\.+(js|jsx)$/,
+          include: [dir],
+          exclude: /node_modules/,
+          use: {
+            loader: 'babel-loader',
+            options: babelLoaderOptions
+          }
+        },
+        {
+          test: /\.+(ts|tsx)$/,
+          include: [dir],
+          exclude: /node_modules/,
+          use: [
+            {
+              loader: 'babel-loader',
+              options: babelLoaderOptions
+            },
+            {
+              loader: 'ts-loader',
+              options: {
+                transpileOnly: true
+              }
+            }
+          ]
+        },
+        {
+          test: /\.css$/,
+          use: cssLoaderConfig()
+        },
+        {
+          test: /\.scss$/,
+          use: cssLoaderConfig('sass-loader')
+        },
+        {
+          test: /\.less$/,
+          use: cssLoaderConfig('less-loader')
+        }
+      ].filter(Boolean)
+    },
+    plugins: [
+      // Useful when profiling
+      // new webpack.ProgressPlugin({
+      //   profile: true
+      // }),
+      // new StatsPlugin(`stats-${isServer ? 'server':'client'}.json`),
+      new webpack.IgnorePlugin(/(precomputed)/, /node_modules.+(elliptic)/),
+      dev && new webpack.NoEmitOnErrorsPlugin(),
+      dev && !isServer && new FriendlyErrorsWebpackPlugin(),
+      dev && new webpack.NamedModulesPlugin(),
+      dev && new webpack.HotModuleReplacementPlugin(), // Hot module replacement
+      dev && new CaseSensitivePathPlugin(), // Since on macOS the filesystem is case-insensitive this will make sure your path are case-sensitive
+      dev && new webpack.LoaderOptionsPlugin({
+        options: {
+          context: dir,
+          customInterpolateName (url, name, opts) {
+            return interpolateNames.get(this.resourcePath) || url
+          }
+        }
+      }),
+      dev && new WriteFilePlugin({
+        exitOnErrors: false,
+        log: false,
+        // required not to cache removed files
+        useHashIndex: false
+      }),
+      !dev && new webpack.IgnorePlugin(/react-hot-loader/),
+      !isServer && !dev && new UglifyJSPlugin({
+        exclude: /react\.js/,
+        parallel: true,
+        sourceMap: false,
+        uglifyOptions: {
+          compress: {
+            comparisons: false
+          }
+        }
+      }),
+      new webpack.DefinePlugin({
+        'process.env.NODE_ENV': JSON.stringify(dev ? 'development' : 'production')
+      }),
+      !isServer && new CombineAssetsPlugin({
+        input: ['manifest.js', 'react.js', 'commons.js', 'main.js'],
+        output: 'app.js'
+      }),
+      !dev && new webpack.optimize.ModuleConcatenationPlugin(),
+      !isServer && extractCSS,
+      !isServer && new PagesPlugin(),
+      !isServer && new DynamicChunksPlugin(),
+      !isServer && new webpack.optimize.CommonsChunkPlugin({
+        name: `commons`,
+        filename: `commons.js`,
+        minChunks: 2
```

Comment:
Where's our previous common chunks logic. We had a much bigger logic than this.

Reply:
Fixed